File: server/app/modules/topics/schema.py
from datetime import datetime

from app.modules.articles.route import ArticleListResponse
from pydantic import BaseModel, Field


class TopicCreate(BaseModel):
    name: str = Field(min_length=1, max_length=100)
    categories: list[str] = Field(default_factory=list)
    tags: list[str] = Field(default_factory=list)


class TopicResponse(TopicCreate):
    id: str
    user_id: str
    is_active: bool
    is_deleted: bool
    created_at: datetime
    updated_at: datetime
    deleted_at: datetime | None = None


class TopicContentResponse(BaseModel):
    articles: ArticleListResponse
    summary: str


# from app.modules.articles.schema import ArticleListResponse
# from pydantic import BaseModel, Field


# class TopicCreate(BaseModel):
#     name: str = Field(
#         min_length=1,
#         max_length=100,
#     )
#     categories: list[str] = Field(default_factory=list)
#     tags: list[str] = Field(default_factory=list)


# class TopicResponse(TopicCreate):
#     id: str
#     user_id: str


# class TopicContentResponse(BaseModel):
#     articles: ArticleListResponse
#     summary: str

File: client/src/features/topics/schema.ts
import { z } from "zod";

import { ArticleListResponseSchema } from "@/features/articles/schema";

export const TopicCreateSchema = z.object({
  name: z.string().min(1).max(100),
  categories: z.array(z.string()).default([]),
  tags: z.array(z.string()).default([]),
});

export const TopicResponseSchema = TopicCreateSchema.extend({
  id: z.string(),
  user_id: z.string(),
  is_active: z.boolean(),
  is_deleted: z.boolean(),
  created_at: z.string(),
  updated_at: z.string(),
  deleted_at: z.string().nullable(),
});

export const TopicContentResponseSchema = z.object({
  articles: ArticleListResponseSchema,
  summary: z.string(),
});

export type TopicCreate = z.infer<typeof TopicCreateSchema>;

export type TopicResponse = z.infer<typeof TopicResponseSchema>;

export type TopicContentResponse = z.infer<typeof TopicContentResponseSchema>;
